plugins/modules/eos_ospfv2.py

```diff
@@ -0,0 +1,1458 @@
+                - id: "0.0.0.50"
+                  range:
+                    address: "172.20.0.0/16"
+                    cost: 34
+              default_information:
+                metric: 100
+                metric_type: 1
+                originate: True
+              distance:
+                intra_area: 85
+              max_lsa:
+                count: 8000
+                ignore_count: 3
+                ignore_time: 6
+                reset_time: 20
+                threshold: 40
+              networks:
+                - area: "0.0.0.0"
+                  prefix: 10.10.2.0/24
+                - area: "0.0.0.0"
+                  prefix: "10.10.3.0/24"
+              redistribute:
+                - routes: "static"
+              router_id: "170.21.0.4"
+      state: rendered
+# "rendered": [
+#         "router ospf 1",
+#         "adjacency exchange-start threshold 20045623",
+#         "area 0.0.0.2 filter 10.1.1.0/24",
+#         "area 0.0.0.50  range 172.20.0.0/16 cost 34",
+#         "default-information originate metric 100 metric-type 1",
+#         "distance ospf intra-area 85",
+#         "max-lsa  8000 40 ignore-count 3  ignore-time 6  reset-time 20",
+#         "network 10.10.2.0/24 area 0.0.0.0",
+#         "network 10.10.3.0/24 area 0.0.0.0",
+#         "redistribute static",
+#         "router-id 170.21.0.4"
+#     ]
+# 
+
+"""
+RETURN = """
+before:
+  description: The configuration prior to the model invocation.
+  returned: always
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+after:
+  description: The resulting configuration model invocation.
+  returned: when changed
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: always
+  type: list
+  sample: ["router ospf 1",
+        "adjacency exchange-start threshold 20045623",
+        "area 0.0.0.2 filter 10.1.1.0/24",
+        "area 0.0.0.50  range 172.20.0.0/16 cost 34",
+        "default-information originate metric 100 metric-type 1",
+        "distance ospf intra-area 85",
+        "max-lsa  8000 40 ignore-count 3  ignore-time 6  reset-time 20",
+        "network 10.10.2.0/24 area 0.0.0.0",
+        "network 10.10.3.0/24 area 0.0.0.0",
+        "redistribute static",
+        "router-id 170.21.0.4"]
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.arista.eos.plugins.module_utils.network.eos.argspec.ospfv2.ospfv2 import (
+    Ospfv2Args,
+)
+from ansible_collections.arista.eos.plugins.module_utils.network.eos.config.ospfv2.ospfv2 import (
+    Ospfv2,
+)
+
+
+def main():
+    """
+    Main entry point for module execution
+
+    :returns: the result form module invocation
+    """
+    module = AnsibleModule(
+        argument_spec=Ospfv2Args.argument_spec, supports_check_mode=True
+    )
```

Comment:
required_if = [
        ("state", "merged", ("config",)),
        ("state", "replaced", ("config",)),
        ("state", "overridden", ("config",)),
        ("state", "parsed", ("running_config",)),
    ]
    mutually_exclusive = [("config", "running_config")]
    module = AnsibleModule(
        argument_spec=Ospfv2Args.argument_spec,
        required_if=required_if,
        supports_check_mode=True,
        mutually_exclusive=mutually_exclusive,
    )
This might help for not config trap